### PR TITLE
Load issues dynamically in the Issue Finder

### LINKED ISSUE #523: Reduce commits by bot on the master branch
## Description
The Creative Commons bot makes a very large number of commits to the `master` branch to keep issues synchronised between GitHub organisation and the issues databag in the repository. This should be reduced as it quickly leads to forks and clones going out of date. Every other copy of the repository goes stale almost every hour.

## Reproduction

1. Fork the repository and wait for a couple of hours.
2. Your repository is now several commits behind `creativecommons:master`.

## Expectation
The commits should either be infrequent, or atomically small, or on a different branch. One of the simplest ways to do that is to eliminate the `updatedAt` field from the data, as even changes to that field (which happen very frequently) lead to new commits. A more long term approach would be to setup serverless functions on Netlify/Vercel to proxy authenticated requests to GitHub on behalf of the client as in [this example](https://github.com/dhruvkb/portfolio-api/blob/master/api/blog_posts/list.js).

## Resolution

- [ ] I would be interested in resolving this bug.

### LINKED ISSUE #542: Migrate to the Search API for the Issue Finder
## Problem
The Issue Finder wanted to use the ["Issues list for org"](https://octokit.github.io/rest.js/v18#issues-list-for-org) API but that required authentication so we built an hourly sync service in `ccos-scripts`.

## Description
Switch to the ["Search issues and pull requests"](https://octokit.github.io/rest.js/v18#search-issues-and-pull-requests) API. This removes the need for hourly sync and eliminates many issues like #523 in the long term.

## Implementation

- [x] I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes
Fixes #542 by @dhruvkb 
Fixes #523 by @dhruvkb 

## Description
This PR uses the Octokit Search API which is not authenticated to find and display issues.

## Technical details
This removes the need for the hourly pull and push in `ccos-scripts` and will also reduce the bot commits to the repo.

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: webpack/js/components.js [hunk 1]
```diff
@@ -178,6 +178,20 @@ export const App = {
     const BASE_URL = 'https://raw.githubusercontent.com/creativecommons/ccos-scripts/master/normalize_repos'
     const FILE_URL = name => `${BASE_URL}/${name}.json`
 
+    const octokit = new Octokit();
+    octokit.search.issuesAndPullRequests({
+      q: 'org:creativecommons is:open is:issue label:"help wanted"',
```
Comment (dhruvkb) [inline webpack/js/components.js:183]: Should we remove the issues awaiting triage?

File: webpack/js/components.js [hunk 2]
```diff
@@ -178,6 +178,20 @@ export const App = {
...
+      per_page: 100
```
Comment (dhruvkb) [inline webpack/js/components.js:184]: I've observed this number to be ~60 so we can skip pagination for now.

**Review by dhruvkb — COMMENTED** [review]
As a side-effect, this fixes the issue of wrong colours being applied to the skill labels.

File: webpack/js/components.js (hunk 1 above)
Comment (kgodey) [reply webpack/js/components.js:183]: I think we should add a separate filter for issues awaiting triage.

**Review by kgodey — COMMENTED** [review]
@dhruvkb the code for this looks fine, but I'm concerned about making this dependent on JavaScript. One of the advantages of having a static site is that it doesn't depend on JavaScript, if we're just using JavaScript for filtering, it can gracefully degrade.

**Comment by dhruvkb** [comment]
@kgodey the content changes so dynamically that we're seeing almost 24 bot commits a day. In cases like this, can we not justify the use of JavaScript?

**Comment by kgodey** [comment]
@dhruvkb I don't see what bot commits have to do with JavaScript or the architecture of the project. In any case, I merged https://github.com/creativecommons/ccos-scripts/pull/94 which should reduce bot commits.

**Comment by dhruvkb** [comment]
@kgodey using JavaScript to fetch issues dynamically eliminates the need to keep a databag and sync it hourly. The data will always be fresh and up-to-date. Adding new filters (such as issues that need triage) is also easier because data is being fetched on demand from GitHub which would otherwise need another databag to be created and pushed hourly.

**Review by akmadian — COMMENTED** [review]
The code here looks great to me, I think I agree with Dhruv on the use of JS. I think that the advantages posed by using JS far outweigh the disadvantages. The JS used seems to be pretty standard so in terms of being reliant on it, it seems pretty innocuous. 

I do agree that I don't like being dependent on third party packages any more than is needed, but the functionality seems worth it to me.

**Review by zackkrida — APPROVED** [review]
I am pleased with the changes and approach taken here. Nice work @dhruvkb

**Comment by dhruvkb** [comment]
Now there's an option to find issues that have the label `🚦 status: awaiting triage`. 

While I agree 100% on @kgodey's points in favour of the static site, this JS based client-side approach proves how easy it is to extend the issue finder when adding new functionality.

**Review by akmadian — APPROVED** [review]

**Review by kgodey — APPROVED** [review]

**Pull request merged**